test/curve_fit.jl

```diff
@@ -1,6 +1,6 @@
 let
     # fitting noisy data to an exponential model
-    model(x, p) = p[1]*exp(-x.*p[2])
+    model(x, p) = @compat p[1] .* exp.((-).(x) .* p[2])
```

Comment:
I think you've overdone it here... it should be sufficient to just vectorize the call to `exp` so it is:
```
p[1] * exp.(-x .* p[2])
```